core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -906,6 +908,28 @@ func (o *evmTxStore) FindReceiptsPendingConfirmation(ctx context.Context, blockN
 	return
 }
 
+// FindTxWithRequestID returns any broadcast ethtx with the given requestID
+func (o *evmTxStore) FindTxWithRequestID(requestID common.Hash) (etx *Tx, err error) {
+	etx = new(Tx)
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		var dbEtx DbEthTx
+		err = tx.Get(&dbEtx, `SELECT * FROM eth_txes WHERE request_id = $1`, requestID)
+		if err != nil {
+			if !errors.Is(err, sql.ErrNoRows) {
+				return pkgerrors.Wrap(err, "FindTxWithRequestID failed to load eth_txes")
+			}
+			return nil
+		}	
+		DbEthTxToEthTx(dbEtx, etx)
```

Comment:
This is new, why is this necessary? Can we not scan directly into the appropriate struct?

Reply:
I ended up taking this from `FindTxWithSequence()` but looks like it has some specific logic around parsing the nonce column into the generic Tx type's sequence field.